### PR TITLE
Expose llvm-foreach --jobs functionality through a driver option

### PR DESCRIPTION
**Pull request by AlexeySachkov**
After https://github.com/intel/llvm/pull/4360 extended `llvm-foreach`
 with an option to specify the desired number of parallel jobs, expose
 this functionality to the DPC++ compiler users.

Co-authored-by: Alexey Sachkov <alexey.sachkov@intel.com>
Co-authored-by: Artem Gindinson <artem.gindinson@intel.com>

### CODE REVIEW

File: clang/include/clang/Driver/Options.td [hunk 1]
```diff
@@ -2680,6 +2680,11 @@ def fsycl_footer_path_EQ : Joined<["-"], "fsycl-footer-path=">,
 def fno_sycl_link_spirv : Flag<["-"], "fno-sycl-link-spirv">,
   Flags<[CoreOption]>, HelpText<"Disable adding of the default (spir64) triple "
   "when discovered in user specified objects and archives.">;
+def fsycl_max_parallel_jobs_EQ : Joined<["-"], "fsycl-max-parallel-link-jobs=">,
+  Flags<[CoreOption]>, Group<f_Group>,
+  HelpText<"Controls the parallelism of post device code link actions performed"
+  "by the toolchain, like generating SPIR-V files for each device image"
+  "or performing an AOT compilation for each device image">;
```
Comment (AGindinson) [inline clang/include/clang/Driver/Options.td:2687]: Do we have to specify the experimental nature of the feature within this help text/in the UsersManual.md block of SYCL documentations?

Comment (mdtoguchi) [reply clang/include/clang/Driver/Options.td:2687]: Yes, I think experimental tags need to be added for both help and documentation.

Comment (AGindinson) [reply clang/include/clang/Driver/Options.td]: Added the experimental marks and improved doc coverage within https://github.com/intel/llvm/pull/4543/commits/e5bb1352de63cf366a6360918e19a69a628f38f1.

File: sycl/doc/UsersManual.md [hunk 2]
```diff
@@ -176,6 +176,15 @@ and not recommended to use in production environment.
     * auto - the compiler will use a heuristic to select the best way of
       splitting device code. This is default mode.
 
+**`-fsycl-max-parallel-link-jobs=<n>`**
+
+    Experimental feature. When specified, it informs the compiler
+    that it can simultaneously spawn up to `N` processes to perform
+    actions required to link DPC++ application. This option is
```
Comment (mdtoguchi) [inline sycl/doc/UsersManual.md:183]:
```suggestion
    actions required to link the DPC++ application. This option is
```

File: sycl/doc/UsersManual.md [hunk 3]
```diff
@@ -176,6 +176,15 @@ and not recommended to use in production environment.
     * auto - the compiler will use a heuristic to select the best way of
       splitting device code. This is default mode.
 
+**`-fsycl-max-parallel-link-jobs=<n>`**
```
Comment (mdtoguchi) [inline sycl/doc/UsersManual.md:179]:
```suggestion
**`-fsycl-max-parallel-link-jobs=<N>`**
```
To match usage in the text body.

File: clang/include/clang/Driver/Options.td [hunk 4]
```diff
@@ -2680,6 +2680,11 @@ def fsycl_footer_path_EQ : Joined<["-"], "fsycl-footer-path=">,
 def fno_sycl_link_spirv : Flag<["-"], "fno-sycl-link-spirv">,
   Flags<[CoreOption]>, HelpText<"Disable adding of the default (spir64) triple "
   "when discovered in user specified objects and archives.">;
+def fsycl_max_parallel_jobs_EQ : Joined<["-"], "fsycl-max-parallel-link-jobs=">,
+  Flags<[CoreOption]>, Group<f_Group>,
+  HelpText<"Experimental feature. Controls the maximum parallelism of actions performed "
```
Comment (mdtoguchi) [inline clang/include/clang/Driver/Options.td:2685]:
```suggestion
  HelpText<"Experimental feature: Controls the maximum parallelism of actions performed "
```
To be consistent with existing Experimental feature tag.

**Review by mdtoguchi — APPROVED** [review]

**Review by bader — APPROVED** [review]

**Pull request merged**